system-int-test/test/cmr/system_int_test/search/granule_shapefile_search_test.clj

```diff
@@ -112,49 +122,85 @@
-                           :content "PROV1"}])]
-              (d/assert-refs-match items found))
+    (testing "GeoJSON Failure cases"
+      (are3 [shapefile additional-params regex]
+        (is (re-find regex
+                      (first (:errors (search/find-refs-with-multi-part-form-post
+                                        :granule
+                                        (let [params [{:name "shapefile"
+                                                       :content (io/file (io/resource (str "shapefiles/" shapefile)))
+                                                       :mime-type "application/geo+json"}]]
+                                          (if (seq additional-params)
+                                            (conj params additional-params)
+                                            params)))))))
+                                  
+        "All granules query"
+        "polygon_with_hole.geojson" {} #"The CMR does not allow querying across granules in all collections with a spatial condition"
 
-        "Single Polygon box around VA and DC"
-        "box.zip" [whole-world very-wide-cart washington-dc richmond]
+        "Failed to parse GeoJSON file"
+        "invalid_json.geojson" {:name "provider" :content "PROV1"} #"Failed to parse GeoJSON file"))
 
-        "Single Polygon box over North pole"
-        "north_pole_poly.zip" [north-pole touches-np on-np whole-world very-tall-cart]
+    (testing "KML Failure cases"
+      (are3 [shapefile additional-params regex]
+        (is (re-find regex
+                     (first (:errors (search/find-refs-with-multi-part-form-post
+                                       :granule
+                                       (let [params [{:name "shapefile"
+                                                      :content (io/file (io/resource (str "shapefiles/" shapefile)))
+                                                      :mime-type "application/vnd.google-earth.kml+xml"}]]
+                                          (if (seq additional-params)
+                                            (conj params additional-params)
+                                            params)))))))
+                                  
+        "All granules query"
+        "polygon_with_hole.kml" {} #"The CMR does not allow querying across granules in all collections with a spatial condition"
 
-        "Single Polygon over Antartica"
-        "antartica.zip" [south-pole touches-sp on-sp whole-world very-tall-cart]
+        "Failed to parse kml file"
+        "invalid.kml" {:name "provider" :content "PROV1"} #"Failed to parse KML file"))
 
-        "Single Polygon over Southern Africa"
-        "southern_africa.zip" [whole-world polygon-with-holes polygon-with-holes-cart normal-line normal-line-cart normal-brs wide-south-cart]
+    (doseq [fmt (keys formats)
+            :let [{extension :extension mime-type :mime-type} (get formats fmt)]]
+      (debug (format "%s %s %s" fmt extension mime-type))
```

Comment:
Similar comments for these two lines from the collection test.

Reply:
Done.